### ISSUE TITLE
Installation issue: E4S project: petsc will not install on Ascent (ORNL) with gcc/6.4.0 

### ISSUE THREAD

**Issue by jgalarowicz** [LLNL/spack#18203]
The package/application turbine fails to build on Ascent (ORNL) with gcc/6.4.0. This is part of the E4S project.
The issue is that the petsc build errors out because the hypre version is too recent.  petsc wants versions less than or equal to 2.18.2 for hypre.   This is the output message.
```
hypre version is 2.19.0 this version of PETSc needs at most 2.18.2
```
We need help from the petsc team to advise on what the course of action should be because the dependencies for hypre are non-trivial in the petsc package.py file.

To recreate the original problem:
spack -e . install --no-cache
with this spack.yaml file in your spack top-level directory:
```
spack:
  packages:
    all:
      providers:
        mpi: [spectrum-mpi]
      target: [ppc64le]
      buildable: true
      version: []
      paths: {}
      modules: {}
      compiler: []
    cuda:
      buildable: false
      version: [10.1.243]
      modules:
        cuda@10.1.243: cuda/10.1.243
      target: []
      providers: {}
      paths: {}
      compiler: []
    spectrum-mpi:
      buildable: false
      version:
      - 10.3.1.2
      modules:
        spectrum-mpi@10.3.1.2: spectrum-mpi/10.3.1.2-20200121
      target: []
      providers: {}
      paths: {}
      compiler: []
    gcc:
      version:
      - 6.4.0
      buildable: true
      target: []
      providers: {}
      paths: {}
      modules: {}
      compiler: []
  specs:
#  - aml
#  - faodel
#  - flecsi+cinch
  - petsc
#  - turbine
#  - unifyfs
  config:
    misc_cache: $spack/cache
    install_tree: $spack/$padding:512
  view: false
  concretization: separately
```
The petsc build fails in this manner:
```
TESTING: checkVersion from config.packages.hypre(/autofs/nccsopen-svm1_home/jgalarowicz/tempspack/spack-stage/spack-stage-petsc-3.13.4-n7izi4sqdhk24vwzxpirfxetd5qkwvxe/spack-src/config/BuildSystem/config/package.py:992)
*******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
hypre version is 2.19.0 this version of PETSc needs at most 2.18.2
Suggest using --download-hypre for a compatible hypre
*******************************************************************************
```

@balay @BarrySmith @jedbrown

**Comment by balay** [comment]
@jgalarowicz this issue should be fixed in #17973

**Comment by jgalarowicz** [comment]
Hi Satish - you are correct.  My repository needed updating to the latest sources.   Petsc does build with the latest spack development branch.   I am closing this issue.   Thank you.   Jim G
